File: mns-core/src/main/java/cz/fb/manaus/spring/CoreLocalConfiguration.java
package cz.fb.manaus.spring;

import com.codahale.metrics.MetricRegistry;
import cz.fb.manaus.core.provider.ExchangeProvider;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Profile;

@Configuration
@ComponentScan(value = "cz.fb.manaus.core")
public class CoreLocalConfiguration {

    @Bean
    public MetricRegistry metricRegistry() {
        return new MetricRegistry();
    }

    @Bean
    @Profile("matchbook")
    public ExchangeProvider matchbookExchangeProvider() {
        return new ExchangeProvider("matchbook", 2d, 1.001d, 0.0075d, false);
    }

    @Bean
    @Profile("betfair")
    public ExchangeProvider betfairExchangeProvider() {
        return new ExchangeProvider("betfair", 2d, 1.01d, 0.065d, true);
    }
}

File: mns-core/src/main/java/cz/fb/manaus/spring/CoreLocalConfiguration.kt
package cz.fb.manaus.spring

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.databind.SerializationFeature
import com.fasterxml.jackson.datatype.jdk8.Jdk8Module
import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
import com.fasterxml.jackson.module.kotlin.KotlinModule
import cz.fb.manaus.core.maintanance.db.OldMarketApprover
import cz.fb.manaus.spring.conf.MarketConf
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.ComponentScan
import org.springframework.context.annotation.Configuration
import org.springframework.http.converter.json.Jackson2ObjectMapperBuilder


@Configuration
@ComponentScan(value = ["cz.fb.manaus.core"])
@EnableConfigurationProperties(MarketConf::class)
open class CoreLocalConfiguration {

    @Bean
    open fun objectMapperBuilder(): Jackson2ObjectMapperBuilder {
        val builder = Jackson2ObjectMapperBuilder()
        builder.modules(KotlinModule(), Jdk8Module(), JavaTimeModule())
                .featuresToDisable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS)
        builder.indentOutput(true)
        return builder
    }

    @Bean
    open fun objectMapper(): ObjectMapper {
        return objectMapperBuilder().build()
    }

    @Bean
    open fun oldMarketApprover(marketConf: MarketConf): OldMarketApprover {
        return OldMarketApprover(marketConf.history)
    }
}
